kano_video/logic/youtube.py

```diff
@@ -45,34 +57,62 @@ def get_last_search_count():
 
 def search_youtube_by_keyword(keyword=None, popular=False, max_results=10,
                               start_index=1, parent_control=False):
-    url = 'http://gdata.youtube.com/feeds/api/videos'
-    params = {
-        'v': 2,
-        'vq': keyword,
-        'racy': 'exclude',
-        'orderby': 'relevance',
-        'alt': 'json',
-        'max-results': max_results,
-        'start-index': start_index
-    }
-    if popular:
-        params['orderby'] = 'viewCount'
 
-    if parent_control is True:
-        params['safeSearch'] = 'strict'
+#----Declare variables -------------
+    global prevPageIndex
 
-    success, error, data = requests_get_json(url, params=params)
+    str_safeSearch="none"
+    str_order = "relevance"
+    str_pageToken = ""
+#-----------------------------------
 
-    if not success:
-        logger.error('Searching YouTube by keyword failed: ' + error)
+
+    youtube = build(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION, #ADDED LINE INIATE OBJECT
+      developerKey=DEVELOPER_KEY)
+
+    if parent_control is True:
+        str_safeSearch = "strict"
+
+    #if popular: #If popular is set it seems like it only returns one video, we will leave that for another day...
+     #   str_order = "viewCount"
+
+#Check to see if we should navigate back or next and set str_pageToken
+    str_pageToken = set_pageToken(start_index)
+    
+  # Call the search.list method to retrieve results matching the specified
+  # query term.
+    data = youtube.search().list(
+      q=keyword,
+      part="id,snippet",
+      maxResults=max_results,
+      safeSearch=str_safeSearch,
+      order=str_order,
+      pageToken=str_pageToken
+    ).execute()
+
+    return_data = []
+
+    # get a list of videos
+    for search_result in data.get("items", []):
+        if search_result["id"]["kind"] == "youtube#video":
+            return_data.append(search_result)
+
+    if return_data == None: #If no data was found then return nothing
+        logger.error('Searching YouTube by keyword failed')
         return None
-    if 'feed' in data and 'entry' in data['feed']:
-        global last_search_count
-        last_search_count = data['feed']['openSearch$totalResults']['$t']
+ 
+    update_pageTokens(data)
 
-        return data['feed']['entry']
+    #Get the total videos found and store it in last_search_count
+    global last_search_count
+    last_search_count = get_total_video_count(data)
+
+    prevPageIndex = start_index # used for page navigation
 
+    return return_data
+        
 
+#TODO UPDATE THIS FUNCTION, THIS DOES NOT APPEAR TO BE IN USE AS OF NOW
 def search_youtube_by_user(username, parent_control=False):
```

Comment:
@cwilson-github Can function comments become function docstrings and in sentence case, i.e. this would become

```python
def search_youtube_by_user(username, parent_control=False):
    """
    TODO: Update this function, this does not appear to be in use as of now
    """

    url = ...
```